### ISSUE TITLE
[PT/TF Adaption] Overfit/Underfit

### ISSUE THREAD

**Issue by astonzhang** [d2l-ai/d2l-en#1105]
For both PyTorch/TensorFlow Adaptations:

In [4.4](http://preview.d2l.ai.s3-website-us-west-2.amazonaws.com/d2l-en/master/chapter_multilayer-perceptrons/underfit-overfit.html), the plots for normal fitting (4.4.4.3.) and overfitting (4.4.4.5) have random non-ideal results: they can significantly differ from the mx plots. 

@AnirudhDagar  @terrytangyuan, can you take a look at them? Or do you have any suggestions about using a different set of hyperparameters (e.g., change `max_degree`) to generate more stable and consistent plots?

**Comment by terrytangyuan** [comment]
Does this occur after reducing the number of epochs?

**Comment by astonzhang** [comment]
Given the current hyperparameters, the current num_epochs is needed to demo underfitting/overfitting.

**Comment by AnirudhDagar** [comment]
@astonzhang I was able to fix the issue after setting seed. @mli @terrytangyuan do you have any other ideas to make this more robust since we won't be using a seed after all.

**Comment by murphyk** [comment]
I noticed this problem, too: I cannot get the SGD optimizer in Pytorch to overfit for the degree-20 polynomial, even when using full-batch training. (I did not try exact least squares fitting.) So this is not a compelling example of overfitting.
My pytorch code is [here](https://github.com/probml/pyprobml/blob/master/book1/supplements/poly_regression_torch.ipynb). By contrast, my [sklearn demo](https://colab.research.google.com/github/probml/pyprobml/blob/master/book1/supplements/linreg_sklearn.ipynb) of overfitting using a similar polynomial regression example clearly shows the problem.
